### DISCUSSION TITLE
How can I add a new section "APP" ?

### DISCUSSION THREAD

**Discussion by xRorkr** [moudey/Shell#257] [Q&A]
Hello, I do not know very well how this works, I'm not a programmer, but I like the application and the ease that brings for some actions, my question is
How can I add a new section "APP" and inside it run applications from my pc ? as steam, winscp and other launcher ? not to have them on my desktop if not in the context menu ?

**Comment by moudey** [comment]
Open `shell.nss` file and add this code to dynamic section
```
menu(type="desktop" title="APP")
{
    item(title="app1" cmd='path/to/app1.exe')
    item(title="app2" cmd='path/to/app2.exe')
    item(title="app3" cmd='path/to/app3.exe')
}
```
Save changes, press <kbd>CTRL</kbd> + <kbd>RIGHT-CLICK</kbd> to reload settings.

**Reply by xRorkr** [reply]
Yes!! ty !! Do you know how I can add a thumbnail for the applications?

**Reply by xRorkr** [reply]
Solved only need "image"

**Reply by moudey** [reply]
Just add the `image` property
```
item(title="test" image cmd='app.exe')

```
or custom image
```
item(title="test" image=#f00 cmd='app.exe')
```

**Reply by xRorkr** [reply]
Ty moudey ❤️last possible question on this topic... adding applications in this menu does not slow down the loading?

**Reply by moudey** [reply]
> Ty moudey ❤️last possible question on this topic... adding applications in this menu does not slow down the loading? 

This is due to the speed of hard disk reading of files that contain icons.

In the first version, it was relatively slow because of loading icons, as a result of loading all menu settings and submenus.  However, since version 1.5, the loading of all submenus settings has been separated separately, and the settings are fetched only when they appear, and this greatly improved the response speed.

**Reply by Eta-Beta** [reply]
Good morning, thank you very much Moudey, by copying the script I managed to create the Compressors section and by replacing "desktop" with "File" the section appears only on files. But now I would also like to make it appear on folders as I do ?. Then I have to move "winrar, 7-zip..... " to the compressors section.

**Reply by moudey** [reply]
add `file` and `dir` to `type` property

```
// in static section
item(type='file|dir' find='"winrar"|"7-zip"' menu="compressors")

// in dynamic section
menu(type='file|dir' title="compressors")
{
}
```

**Reply by Eta-Beta** [reply]
Trying and trying again, I always managed to have your scripts that you gave me in other discussions to create the section and transfer 7-Zip........ , but not in the folder menu. Thanks for the umpteenth script (but does "dir" indicate folders?). Another thing to put the icon of the section I have to write "menu(type='file|dir' title="Compressors" image??)" where do I get the icons?

**Reply by moudey** [reply]
>but does "dir" indicate folders?

dir = directory

>where do I get the icons?

https://nilesoft.org/gallery/glyphs

**Reply by Eta-Beta** [reply]
Ok everything works, thanks as always for the explanations. Thanks for the link to the icons (I could have looked it up, sorry). One last question, can't putting an icon like Winrar for example be done?

**Reply by moudey** [reply]
```
image='path/to/icon'

image='path/to/dll_or_exe,0'
```

https://nilesoft.org/docs/functions/image

**Reply by Eta-Beta** [reply]
Thanks and ..... maybe better read me all the documentation before asking questions. Thank you always for your availability, see you next time.

**Comment by Eta-Beta** [comment]
Good morning Moudey, I'm re-editing this post due to a problem relating to the changes made thanks to your work. Thanks to you I have my "COMPATTATORI" section where I have moved the software to compress the files (as you can see from the screenshot). The problem arises when I want to select many files to make a single archive, the section does not appear (as can be seen from the second screenshot). Can you kindly tell me what changes I need to make so that the "COMPATTATORI" section appears?. Thanks in advance for your help.

![Screenshot 2023-11-16 111014](https://github.com/moudey/Shell/assets/104278263/17c5e840-671b-4195-9a25-1946f58167bc)


![Screenshot 2023-11-16 111101](https://github.com/moudey/Shell/assets/104278263/23e22599-b815-44c3-9cbc-3502561c34b9)

**Comment by RubicBG** [comment]
menu(mode='multiple' title='COMPATTATORI') {}

**Reply by Eta-Beta** [reply]
Hello and thank you very much, I didn't think so little would be enough, now it works perfectly.
